src/pam_hbac_obj.c

```diff
@@ -172,6 +172,27 @@ get_user_groups(const char *name, gid_t primary_gid,
         ret = 0;
         *ngroups_ptr = ngroups;
     }
+#elif defined(HAVE_GETGRSET)
+    int ngroups;
+    long max_group_len;
+    char *gid_list_s, *gid_s;
+
+    ngroups = 0;
+    max_group_len = sysconf(_SC_LOGIN_NAME_MAX);
+
+    /* string containing comma separated list of gids the user belongs to */
+    gid_list_s = malloc(sizeof(char)*max_group_len*NGROUPS_MAX);
+    gid_list_s = getgrset(name);
```

Comment:
The memory pointed to by the gid_list_s pointer is lost, I think `free()` needs to be called just before `#else`.

Reply:
The link to the documentation you posted says that it should be freed yes.